zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -316,8 +312,10 @@ where
             reconnect
         };
 
-        // SECURITY: rate-limit new candidate connections
-        sleep.await;
+        // SECURITY: rate-limit new outbound peer connections
+        (&mut self.wait_next_handshake).await;
+        let mut sleep = sleep(constants::MIN_PEER_CONNECTION_INTERVAL);
+        mem::swap(&mut self.wait_next_handshake, &mut sleep);
```

Comment:
Two fixes:
- only wait if we're returning `Some`
- schedule the next sleep time after the previous sleep returns, avoiding issues with overlapping times